trieste/space.py

```diff
@@ -501,13 +501,21 @@ def __repr__(self) -> str:
     def lower(self) -> TensorType:
         """The lowest values taken by each space dimension, concatenated across subspaces."""
         lower_for_each_subspace = [self.get_subspace(tag).lower for tag in self.subspace_tags]
-        return tf.concat(lower_for_each_subspace, axis=-1)
+        return (
+            tf.concat(lower_for_each_subspace, axis=-1)
+            if lower_for_each_subspace
+            else tf.constant([], dtype=tf.float64)
```

Comment:
could there be a problem bcs of setting the dtype to some default, can we infer it from some object here? 
(in some cases, e.g. keras models, we would want to instantiate float32 search space, would this make them issues for models where we use float32?)

Reply:
We already default to tf.float64 in Box when we pass in a list of floats (including an empty one), so at least we're consistent. There's probably also a workaround by using a zero-dimension Box (defined with tf.constants) instead of a zero sequence tagged product space. Suggest we don't worry about it unless we hit an actual issue.

Reply:
ok, what about the following small improvement - declaring a constant dtype variable somewhere at the top and using it throughout?

Reply:
Done